File: tuts/tut13/frequencies.py
#!/usr/bin/python3

import sys
import re

lett_map = {}
for line in sys.stdin:
    line = re.sub(r'[^a-zA-Z0-9]','',line)
    for letter in line:
        if letter not in lett_map:
            lett_map[letter] = 1
        else:
            lett_map[letter] += 1


for letter in sorted(lett_map):
    print("'"+letter+"' occured", lett_map[letter], "times")

File: tuts/tut13/frequencies.pl
#!/usr/bin/perl

my %lett_map;
while ($line = <STDIN>) {
	chomp $line;
	$line =~ s/[^a-zA-Z0-9]//g;	
	@letters = split //, $line;
	for $letter (@letters) {
		if (exists $lett_map{$letter}) {
			$lett_map{$letter} += 1;					
		} else {		
			$lett_map{$letter} = 1;					
		}

	}
}

for $l (sort(keys %lett_map)) {
	print "'$l' has occured $lett_map{$l} times\n"
}
